src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -24,6 +24,62 @@ private struct DateTimeParseData
             public byte OffsetToken;
         }
 
+        public static string FormatDateTimeOffset(DateTimeOffset value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeOffsetTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
+        }
+
+        public static string FormatDateTime(DateTime value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
+        }
+
+        public static bool TryParseAsISO(ReadOnlySpan<char> source, out DateTime value)
+        {
+            if (!JsonReaderHelper.IsValidDateTimeOffsetParseLength(source.Length))
+            {
+                value = default;
+                return false;
+            }
+
+            int length = JsonReaderHelper.GetUtf8ByteCount(source);
```

Comment:
Generally speaking, the length of `ReadOnlySpan<char> source` should always be relatively small, correct?

In fact, the above check using `IsValidDateTimeOffsetParseLength` guarantees its between 10 and 6*(33+16-7), i.e. 252.

So, could we always just alloc on the stack using `JsonConstants.StackallocThreshold` (with `Debug.Assert` that `JsonReaderHelper.GetUtf8ByteCount(source)` <= `JsonConstants.StackallocThreshold`)?

That way, we don't have to call the `GetUtf8ByteCount` method to actually get the exact count.